### ISSUE TITLE
Apache2 proxy forwarding to unix sockets with mod_proxy_uwsgi

### ISSUE THREAD

**Issue by mostarman** [unbit/uwsgi#890]
Hi - I am having several issues trying to set up an Apache2 web server (version 2.4.12) as a proxy forwarding requests to a unix socket using the mod_proxy_uwsgi module.
I am loading the module like this: ProxyPass / unix:/tmp/uwsgi.sock|uwsgi://
So first there are a few iffy lines in the uwsgi_canon function (like char *host and sport never being initialized), but even with that fixed I can never get it to recognize unix: as a web socket - it always tries to connect to host:sport. Am I missing something?

**Comment by unbit** [comment]
The syntax for unix sockets changed during 2.4 development cycle, check if the mod_proxy_fcgi approach works in latest versions: http://httpd.apache.org/docs/2.4/mod/mod_proxy_fcgi.html

**Comment by niol** [comment]
Same here, I am using ```ProxyPass "/r/" "unix:/var/run/uwsgi/foo.socket|uwsgi://localhost/"``` and the patch below which avoids using the unitialized ```char* host``` but it still does not want to connect via unix domain socket (UDS): ```ap_proxy_determine_connection``` does not set ```uds_path``` for some reason. I'll follow up if  I find something.

```diff
--- uwsgi-2.0.7.orig/apache2/mod_proxy_uwsgi.c
+++ uwsgi-2.0.7/apache2/mod_proxy_uwsgi.c
@@ -67,19 +67,14 @@ static int uwsgi_canon(request_rec *r, c
     }
     url += sizeof(UWSGI_SCHEME); /* Keep slashes */

-    // is it a unix socket ?
-    if (strlen(url) == 2) {
-       *sport = 0;
-    }
-    else {
-        err = ap_proxy_canon_netloc(r->pool, &url, NULL, NULL, &host, &port);
-        if (err) {
-            ap_log_rerror(APLOG_MARK, APLOG_ERR, 0, r,
-                      "error parsing URL %s: %s", url, err);
-            return HTTP_BAD_REQUEST;
-        }
-       apr_snprintf(sport, sizeof(sport), ":%u", port);
+    err = ap_proxy_canon_netloc(r->pool, &url, NULL, NULL, &host, &port);
+    if (err) {
+        ap_log_rerror(APLOG_MARK, APLOG_ERR, 0, r,
+                    "error parsing URL %s: %s", url, err);
+        return HTTP_BAD_REQUEST;
     }
+    ap_log_rerror(APLOG_MARK, APLOG_ERR, 0, r, "parsed to: %s:%d", host, port);
+    apr_snprintf(sport, sizeof(sport), ":%u", port);

     if (ap_strchr(host, ':')) { /* if literal IPv6 address */
         host = apr_pstrcat(r->pool, "[", host, "]", NULL);
```

(this was also reported as https://github.com/unbit/uwsgi/issues/973)

**Comment by kontza** [comment]
Ditto here. My stabs at trying to get this thing working was to copy the canon-function from mod_proxy_cgi, strip out one struct (`fcgi_req_config_t`), and change schemes from `fcgi`to `uwsgi`. This got me going into a promising direction: I ended up with permission denied. Then I remembered, that  `uwsgi` was started with my own uid, and apache2 us running as _www (I'm on OS X). I restarted `uwsgi` as _www, and the result was `AH00898: DNS lookup failure for:  returned by /tuts`

My ProxyPass is as follows
```
ProxyPass /tuts unix:/tmp/uwsgi-tutorial.sock|uwsgi://
```

This is as far as I will go with this, since this is not my core task at the moment, I'll revert back to using IP:PORT -sockets. This is because the servers I will be deploying to run Centos 7, and I just cannot go around and install custom mod_proxy_uwsgi-modules here and there.
